jar-service/src/main/java/com/sixsq/slipstream/application/CommonStatusService.java

```diff
@@ -136,10 +158,20 @@ private Representation toJson(Status status) {
 		StringBuilder json = new StringBuilder();
 
 		json.append("{\n");
-		json.append("   \"error\": \"" + status.getCode() + "\",\n");
-		json.append("   \"reason\": \"" + status.getReasonPhrase() + "\",\n");
-		json.append("   \"detail\": \"" + status.getDescription() + "\"\n");
-		json.append("}\n");
+		json.append("    \"error\": {\n");
+		json.append("        \"code\": \"" + status.getCode() + "\"");
+
+		if (status.getReasonPhrase() != null)
+			json.append(",\n        \"reason\": \"" + status.getReasonPhrase() + "\"");
+
+		if (status.getDescription() != null)
+			json.append(",\n        \"detail\": \"" + status.getDescription() + "\"");
+
+		if (status.getUri() != null)
+			json.append(",\n        \"specificationUri\": \"" + status.getUri() + "\"");
+
+		json.append("\n    }\n");
+		json.append("}");
 
```

Comment:
This is my "final" goal but currently there is not support for JSON is SS Server.
So my idea was to wait for the `json` branch to be merged and then use the same serializer to generate this error.